conda_build/utils.py

```diff
@@ -1175,12 +1175,74 @@ def package_has_file(package_path, file_path, refresh=False):
 
 
 def ensure_list(arg):
-    if (isinstance(arg, string_types) or not hasattr(arg, '__iter__')):
-        if arg is not None:
-            arg = [arg]
-        else:
-            arg = []
-    return arg
+    """
+    Ensure the object is a list. If not return it in a list.
+
+    :param arg: Object to ensure is a list
+    :type arg: any
+    :return: `arg` as a `list`
+    :rtype: `list`
+    """
+    if arg is None:
+        return []
+    elif islist(arg):
+        return arg
```

Comment:
So I'm returning any iterable here (which is what the old `ensure_list` also did), this results in a number of places where we end up doing `list(ensure_list(obj))` to avoid generator issues and the likes.

Do we want to simply wrap this in `list` and just get it over with?

Doing so was also proposed in PR #3979.

Reply:
Sounds fine to me.